### PR TITLE
Add Before & After TextChangeEvent bindings to RxTextView

### LINKED ISSUE #137: RxTextView.textChangesWithBefore
New to RxJava and RxBinding, I started out by doing some user input validation previously implemented via `TextWatcher`. I realized while attempting to leverage the `filter` operator that the Observable returned from `RxTextView.textChanges` only emits the new text. Would it be beneficial to add a similar Observable that also emitted the before value (from [`TextWatcher#beforeTextChanged`](https://developer.android.com/reference/android/text/TextWatcher.html#beforeTextChanged%28java.lang.CharSequence,%20int,%20int,%20int%29))?

### PR DESCRIPTION
**Pull request by xfumihiro**
Closes #137.

### CODE REVIEW

**Comment by xfumihiro** [comment]
Hmm... tests have passed on my machine, though.

File: rxbinding/src/main/java/com/jakewharton/rxbinding/widget/TextViewBeforeTextChangeEvent.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import android.widget.TextView;
+import com.jakewharton.rxbinding.view.ViewEvent;
+
+/**
+ * A before text-change event on a view.
+ * <p>
+ * <strong>Warning:</strong> Instances keep a strong reference to the view. Operators that cache
+ * instances have the potential to leak the associated {@link Context}.
+ */
+public class TextViewBeforeTextChangeEvent extends ViewEvent<TextView> {
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding/widget/TextViewBeforeTextChangeEvent.java]: +`final`

File: rxbinding/src/main/java/com/jakewharton/rxbinding/widget/TextViewAfterTextChangeEvent.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import android.widget.TextView;
+import com.jakewharton.rxbinding.view.ViewEvent;
+
+/**
+ * An after text-change event on a view.
+ * <p>
+ * <strong>Warning:</strong> Instances keep a strong reference to the view. Operators that cache
+ * instances have the potential to leak the associated {@link Context}.
+ */
+public class TextViewAfterTextChangeEvent extends ViewEvent<TextView> {
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding/widget/TextViewAfterTextChangeEvent.java]: +`final`

File: rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxTextViewTest.java [hunk 3]
```diff
@@ -119,6 +130,102 @@
...
+    assertThat(event2.view()).isSameAs(view);
+    assertThat(event2.editable().toString()).isEqualTo("He");
+
+    subscription.unsubscribe();
+
+    view.setText("Silent");
+    o.assertNoMoreEvents();
+  }
+
+  @Test @UiThreadTest public void textWatchEvents() {
```
Comment (JakeWharton) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxTextViewTest.java]: What is this testing that the other, individual tests do not cover?

Comment (xfumihiro) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxTextViewTest.java]: Should be nothing, but the test failed on API 18 and passed on API 21...

Comment (xfumihiro) [inline rxbinding/src/androidTest/java/com/jakewharton/rxbinding/widget/RxTextViewTest.java]: Will remove it.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**